samples/snippets/cloud-client/src/main/java/privateca/DeleteCertificateAuthority.java

```diff
@@ -0,0 +1,109 @@
+    // caPoolName: The name of the CA pool under which the CA is present.
+    // certificateAuthorityName: The name of the CA to be deleted.
+    String project = "your-project-id";
+    String location = "ca-location";
+    String caPoolName = "ca-pool-name";
+    String certificateAuthorityName = "certificate-authority-name";
+    deleteCertificateAuthority(project, location, caPoolName, certificateAuthorityName);
+  }
+
+  // Delete the Certificate Authority from the specified CA pool.
+  // Before deletion, the CA must be disabled and must not contain any active certificates.
+  public static void deleteCertificateAuthority(
+      String project, String location, String caPoolName, String certificateAuthorityName)
+      throws IOException, ExecutionException, InterruptedException {
+    // Initialize client that will be used to send requests. This client only needs to be created
+    // once, and can be reused for multiple requests. After completing all of your requests, call
+    // the `certificateAuthorityServiceClient.close()` method on the client to safely
+    // clean up any remaining background resources.
+    try (CertificateAuthorityServiceClient certificateAuthorityServiceClient =
+        CertificateAuthorityServiceClient.create()) {
+      // Create the Certificate Authority Name.
+      CertificateAuthorityName certificateAuthorityNameParent =
+          CertificateAuthorityName.newBuilder()
+              .setProject(project)
+              .setLocation(location)
+              .setCaPool(caPoolName)
+              .setCertificateAuthority(certificateAuthorityName)
+              .build();
+
+      // Check if the CA is enabled.
+      if (certificateAuthorityServiceClient
+              .getCertificateAuthority(certificateAuthorityNameParent)
+              .getState()
+          == State.ENABLED) {
+        System.out.println("Please disable the Certificate Authority before deletion !");
+        return;
+      }
+
+      // Create the DeleteCertificateAuthorityRequest.
+      // Setting the setIgnoreActiveCertificates() to true, will delete the CA
+      // even if it contains active certificates. Care should be taken to re-anchor
+      // the certificates to new CA before deleting.
+      DeleteCertificateAuthorityRequest deleteCertificateAuthorityRequest =
+          DeleteCertificateAuthorityRequest.newBuilder()
+              .setName(certificateAuthorityNameParent.toString())
+              .setIgnoreActiveCertificates(false)
+              .build();
+
+      // Delete the Certificate Authority.
+      ApiFuture<Operation> futureCall =
+          certificateAuthorityServiceClient
+              .deleteCertificateAuthorityCallable()
+              .futureCall(deleteCertificateAuthorityRequest);
+      Operation response = futureCall.get();
+
+      if (response.hasError()) {
+        System.out.println("Error while deleting Certificate Authority !");
```

Comment:
I would print out the error here as well for easier debugging.

Reply:
Thank you for the suggestion. Incorporated it.